### PR TITLE
fix(semantics): fix incorrect type checking that prevented `IpAddr in

### PR DESCRIPTION
**Pull request by dndx**
IpCidr` matchings

https://konghq.atlassian.net/browse/KAG-1846

### CODE REVIEW

File: src/semantics.rs [hunk 1]
```diff
@@ -123,11 +127,11 @@ impl Validate for Expression {
                         }
                     },
                     BinaryOperator::In | BinaryOperator::NotIn => {
-                        match p.rhs {
-                            Value::String(_) | Value::IpCidr(_) => {
+                        match (lhs_type, &p.rhs,) {
```
Comment (chronolaw) [inline src/semantics.rs:130]: Should be `(lhs_type, &p.rhs)`? (remove `,`)?

Comment (dndx) [reply src/semantics.rs:130]: This is by auto format, and is the recommended format for tuples by Rustfmt.

File: t/07-in_notin.t [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+nilIn/NotIn operators only supports string in string or IP in CIDR
+nilIn/NotIn operators only supports string in string or IP in CIDR
+--- no_error_log
+[error]
+[warn]
+[crit]
+
+
+
+=== TEST 2: in operator works with String operands
```
Comment (chronolaw) [inline t/07-in_notin.t:59]: We should add `notin` test case here.

File: t/07-in_notin.t [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            local schema = require("resty.router.schema")
+            local router = require("resty.router.router")
+            local context = require("resty.router.context")
+
+            local s = schema.new()
+
+            s:add_field("l3.ip", "IpAddr")
+
+            local r = router.new(s)
+            assert(r:add_matcher(0, "a921a9aa-ec0e-4cf3-a6cc-1aa5583d150c",
```
Comment (chronolaw) [inline t/07-in_notin.t:119]: We should add `notin` test case here.

Comment (dndx) [reply t/07-in_notin.t:76]: Added

**Review by chronolaw — APPROVED** [review]

**Pull request merged**